src/descrambler/descrambler.c

```diff
@@ -1026,7 +1026,7 @@ descrambler_descramble ( service_t *t,
     tk = (tsb[3] & 0x80) != 0 ? key_find_struct(dr, tsb, t) : dr->dr_key_last;
   }  
 
-  if (dr->dr_queue_total == 0 && (tsb[3] & 0x80) == 0) {
+  if (dr->dr_queue_total == 0 && (tsb[3] & 0x80) != 0) {
```

Comment:
This is wrong. It would reorder packets. If the descrambling is activated, all packets (including not scrambled ones) must go through the descrambling routines.